src/com/google/enterprise/adaptor/fs/FileDelegate.java

```diff
@@ -205,4 +206,33 @@ void startMonitorPath(Path watchPath, AsyncDocIdPusher pusher)
    * terminating any file system change monitors.
    */
   void destroy();
+
+  /**
+   * Helper class for wrapping a collection of {@link Path} as a
+   * {@link DirectoryStream}.
+   */
+  abstract static class PathDirectoryStream
+      implements DirectoryStream<Path> {
+    private final Iterable<Path> paths;
+    private boolean mayGetIterator = true;
+
+    public abstract Iterable<Path> getPaths() throws IOException;
+
+    PathDirectoryStream() throws IOException {
+      paths = getPaths();
+    }
+
+    @Override
+    public Iterator<Path> iterator() {
+      Preconditions.checkState(mayGetIterator,
+          "DirectoryStream can only have one iterator.");
```

Comment:
From the DirectoryStream javadoc:

While DirectoryStream extends Iterable, it is not a general-purpose Iterable as it supports only a single Iterator; invoking the iterator method to obtain a second or subsequent iterator throws IllegalStateException.